app/src/test/java/io/hstream/testing/TestUtils.java

```diff
@@ -199,9 +197,12 @@ public static boolean isAscending(List<RecordId> input) {
   }
 
   public static void assertRecordIdsAscending(List<ReceivedRawRecord> input) {
-    Assertions.assertTrue(
-        isAscending(
-            input.stream().map(ReceivedRawRecord::getRecordId).collect(Collectors.toList())));
+    if (!input.isEmpty()) {
```

Comment:
no need to check here?  call map on an empty list should return an empty list and will be rejected in `isAscending()`  method

Reply:
if you want `assertRecordIdsAscending(emptyList)` success, then modify the implemention of `isAscending()` method.